src/lib/stitching/exchange/__tests__/stitching.test.ts

```diff
@@ -0,0 +1,121 @@
+const restOfResolveArgs = {
+  operation: "query",
+  schema: expect.anything(),
+  context: expect.anything(),
+  transforms: expect.anything(),
+  info: expect.anything(),
+}
+
+describe("when handling resolver delegation", () => {
+  it("requests an artwork from an LineItem's artworkId", async () => {
+    const { resolvers } = await getExchangeStitchedSchema()
+    const artworkResolver = resolvers.CommerceLineItem.artwork.resolve
+    const mergeInfo = { delegateToSchema: jest.fn() }
+
+    artworkResolver({ artworkId: "ARTWORK-ID" }, {}, {}, { mergeInfo })
+
+    expect(mergeInfo.delegateToSchema).toHaveBeenCalledWith({
+      args: { id: "ARTWORK-ID" },
+      fieldName: "artwork",
+      ...restOfResolveArgs,
+    })
+  })
+
+  it("calls a user or partner when looking up party details ", async () => {
+    const { resolvers } = await getExchangeStitchedSchema()
+    const { buyerDetails } = resolvers.CommerceBuyOrder
+    const info = { mergeInfo: { delegateToSchema: jest.fn() } }
+
+    info.mergeInfo.delegateToSchema.mockResolvedValue({})
+
+    const parentUser = {
+      buyerDetails: { __typename: "CommerceUser", id: "USER-ID" },
+    }
+
+    buyerDetails.resolve(parentUser, {}, {}, info)
+
+    expect(info.mergeInfo.delegateToSchema).toHaveBeenCalledWith({
+      args: { id: "USER-ID" },
+      operation: "query",
+      fieldName: "user",
+      ...restOfResolveArgs,
+    })
+
+    // Reset and verify what happens when we get a partner's details
+    // back from Exchange
+    info.mergeInfo.delegateToSchema.mockReset()
+    info.mergeInfo.delegateToSchema.mockResolvedValue({})
+
+    const parentPartner = {
+      buyerDetails: { __typename: "CommercePartner", id: "PARTNER-ID" },
+    }
+
+    buyerDetails.resolve(parentPartner, {}, {}, info)
+
+    expect(info.mergeInfo.delegateToSchema).toHaveBeenCalledWith({
+      args: { id: "PARTNER-ID" },
+      operation: "query",
+      fieldName: "partner",
+      ...restOfResolveArgs,
+    })
+  })
+})
+
+it("delegates to the local schema for an LineItem's artwork", async () => {
+  const { resolvers } = await getExchangeStitchedSchema()
+  const artworkResolver = resolvers.CommerceLineItem.artwork.resolve
+  const mergeInfo = { delegateToSchema: jest.fn() }
+
+  artworkResolver({ artworkId: "ARTWORK-ID" }, {}, {}, { mergeInfo })
+
+  expect(mergeInfo.delegateToSchema).toHaveBeenCalledWith({
+    args: { id: "ARTWORK-ID" },
+    fieldName: "artwork",
+    ...restOfResolveArgs,
+  })
+})
+
+it("delegates to the right schema when ", async () => {
+  const { resolvers } = await getExchangeStitchedSchema()
+  const artworkResolver = resolvers.CommerceLineItem.artwork.resolve
+  const mergeInfo = { delegateToSchema: jest.fn() }
+
+  artworkResolver({ artworkId: "ARTWORK-ID" }, {}, {}, { mergeInfo })
+
+  expect(mergeInfo.delegateToSchema).toHaveBeenCalledWith({
+    args: { id: "ARTWORK-ID" },
+    operation: "query",
+    fieldName: "artwork",
+    ...restOfResolveArgs,
+  })
+})
```

Comment:
I have a bit of a hard time reading these tests and their stated behavioural intent. Is it possible to make these actually work without spying on `delegateToSchema`? E.g. by providing a mocked exchange schema (using grapqhl-tools like we do in Reaction/Emission)? If not, then at least moving some of the boilerplate setup out of the tests might make it easier to focus on the core parts of the tests.